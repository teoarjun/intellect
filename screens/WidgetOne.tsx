import * as React from 'react';
import {StyleSheet, View, Text, Button, TouchableOpacity} from 'react-native';
import QuestionScreenStyle from '../assets/styles/QuestionScreen';
import Slider from '@react-native-community/slider';
import * as Colors from '../assets/styles/Colors';
import ProgressCircle from 'react-native-progress-circle';
import Icon from 'react-native-vector-icons/Ionicons';

const WidgetOne = (props: any) => {
  const [answer, setAnswer] = React.useState(5);
  const moveToScreen = () => {
    /* istanbul ignore next */
    props.navigation.navigate('Home');
  };
  const moveToTwo = () => {
    /* istanbul ignore next */
    props.navigation.navigate('WidgetTwo');
  };

  const buttons = () => {
    return (
      <View style={styles.nextButton}>
        <Button title="Next" color="#000000" onPress={moveToTwo} />
      </View>
    );
  };
  const renderSlider = () => {
    return <View style={styles.questionRow}>
    <Slider
      minimumValue={0}
      value={answer}
      maximumValue={10}
      tapToSeek={true}
      onValueChange={value => {
        /* istanbul ignore next */
        setAnswer(Math.round(value))
      }}
      thumbTintColor={Colors.ACTIVE_COLOR}
      minimumTrackTintColor={Colors.ACTIVE_COLOR}
      maximumTrackTintColor={Colors.WHITE}
    />
  </View>
  }
  const renderCircle = () => {
    return <View style={styles.circleContainer}>
    <View style={styles.circleSlider}>
      <View style={styles.childCircle}>
        <ProgressCircle
          percent={Math.round(answer) * 10}
          radius={50}
          borderWidth={8}
          color="#FFFFFF"
          shadowColor={Colors.ACTIVE_COLOR}
          bgColor="red">
          <View style={styles.smallestCircle}>
            <Text style={styles.circleText}>{answer}</Text>
          </View>
        </ProgressCircle>
      </View>
    </View>
  </View>;
  }

  return (
    <View style={styles.container} testID="calendar">
      <View style={styles.progressRow}>
        <View style={styles.activeColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
        <View style={styles.inActiveColor} />
      </View>
      <View style={styles.questionNoRow}>
        <View style={styles.center}>
          <Text style={styles.questionText}>
            Rescue Session: Anger and Frustration
          </Text>
        </View>
        <TouchableOpacity style={styles.center} onPress={moveToScreen}>
          <Icon name="close" size={30} />
        </TouchableOpacity>
      </View>

      <View style={styles.titleRow}>
        <Text style={styles.title}>
          Pick your level of anger and frustration?
        </Text>
      </View>

      {renderCircle()}

      {renderSlider()}

      {buttons()}
    </View>
  );
};

const styles = StyleSheet.create(QuestionScreenStyle);

export default WidgetOne;
