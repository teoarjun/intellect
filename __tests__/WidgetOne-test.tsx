// Mock out all top level functions, such as get, put, delete and post:
import React from 'react';
import renderer from 'react-test-renderer';
import WidgetOne from '../screens/WidgetOne';

jest.mock('react-native-progress-circle', () => 'ProgressCircle');
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

it('renders correctly with defaults', () => {
  const navigation = {navigate: jest.fn()};
  const button = renderer
    .create(<WidgetOne name="World" navigation={navigation} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
