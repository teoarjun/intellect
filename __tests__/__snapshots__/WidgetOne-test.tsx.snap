// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#96ccd5",
      "flex": 1,
      "padding": 20,
      "paddingTop": 50,
    }
  }
  testID="calendar"
>
  <View
    style={
      Object {
        "backgroundColor": "#96ccd5",
        "borderRadius": 20,
        "flexDirection": "row",
        "height": 7,
        "marginVertical": 10,
        "maxWidth": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#5DABBA",
          "borderRadius": 20,
          "marginHorizontal": 5,
          "width": "18%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "textTransform": "uppercase",
          }
        }
      >
        Rescue Session: Anger and Frustration
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    >
      <Icon
        name="close"
        size={30}
      />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Pick your level of anger and frustration?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#000000",
          "borderRadius": 100,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "height": 200,
          "justifyContent": "center",
          "padding": 20,
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5DABBA",
            "borderColor": "rgba(0, 0, 0, 0.08)",
            "borderRadius": 75,
            "borderTopWidth": 0,
            "height": 150,
            "justifyContent": "center",
            "padding": 20,
            "width": 150,
          }
        }
      >
        <ProgressCircle
          bgColor="red"
          borderWidth={8}
          color="#FFFFFF"
          percent={50}
          radius={50}
          shadowColor="#5DABBA"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6DA18F",
                "borderColor": "rgba(0, 0, 0, 0.08)",
                "borderRadius": 60,
                "borderWidth": 1,
                "height": 120,
                "justifyContent": "center",
                "padding": 20,
                "width": 120,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 40,
                }
              }
            >
              5
            </Text>
          </View>
        </ProgressCircle>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#ffffff"
      maximumValue={10}
      minimumTrackTintColor="#5DABBA"
      minimumValue={0}
      onChange={[Function]}
      onRNCSliderSlidingComplete={null}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={0}
      style={
        Object {
          "height": 40,
        }
      }
      tapToSeek={true}
      thumbImage={null}
      thumbTintColor="#5DABBA"
      value={5}
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 30,
        "bottom": 20,
        "left": 20,
        "marginVertical": 20,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
